pymeasure/instruments/hp/hpsystempsu.py

```diff
@@ -0,0 +1,330 @@
+            self.status_bytes_count, "little"))
+        return self.status_bits.from_buffer(reply)
+
+    id = HPLegacyInstrument.measurement(
+        "ID?",
+        """
+        Reads the ID of the instrument and returns this value for now
+
+        """,
+    )
+
+    current = HPLegacyInstrument.control(
+        "IOUT?", "ISET %g",
+        """
+        A floating point property that controls the output current of the device.
+
+        """,
+        validator=strict_range,
+        values=[0, limits["HP6632A"]["Cur_lim"]],
+    )
+
+    over_voltage_limit = HPLegacyInstrument.setting(
+        "OVSET %g",
+        """
+        A floationg point property that sets the OVP threshold.
+
+        """,
+        validator=strict_range,
+        values=[0, limits["HP6632A"]["OVP_lim"]],
+    )
+
+    OCP_enabled = HPLegacyInstrument.setting(
+        "OCP %d",
+        """
+        A bool property which controls if the OCP (OverCurrent Protection) is enabled
+        """,
+        validator=strict_discrete_set,
+        values={False: 0, True: 1},
+        map_values=True
+    )
+
+    output_enabled = HPLegacyInstrument.setting(
+        "OUT %d",
+        """
+        A bool property which controls if the outputis enabled
+        """,
+        validator=strict_discrete_set,
+        values={False: 0, True: 1},
+        map_values=True
+    )
+
+    @output_enabled.getter
+    def output_enabled(self):
+        """
+        A bool property which controls if the output is enabled
+        """
+        output_status = bool(self.status.CV or self.status.CCpos or
+                             self.status.CCneg or self.status.Unregulated)
+        return output_status
+
+    def reset_OVP_OCP(self):
+        """
+        Resets Overvoltage and Overcurrent protections
+
+        """
+        self.write("RST")
+
+    rom_Version = HPLegacyInstrument.measurement(
+        "ROM?",
+        """
+        Reads the ROM id (software version) of the instrument and returns this value for now
+
+        """,
+    )
+
+    SRQ_enabled = HPLegacyInstrument.setting(
+        "SRQ %d",
+        """
+        A bool property which controls if the SRQ (ServiceReQuest) is enabled
+        """,
+        validator=strict_discrete_set,
+        values={False: 0, True: 1},
+        map_values=True
+    )
+
+    voltage = HPLegacyInstrument.control(
+        "VOUT?", "VSET %g",
+        """
+        A floating point proptery that controls the output voltage of the device.
+
+        """,
+        validator=strict_range,
+        values=[0, limits["HP6632A"]["Volt_lim"]],
+    )
+
+
+class HP6633A(HP6632A):
+    """ Represents the Hewlett Packard 6633A system power supply
+    and provides a high-level interface for interacting
+    with the instrument.
+    """
+
+    name = "Hewlett Packard HP6633A"
+    current_values = [0, limits["HP6633A"]["Cur_lim"]]
+    OVP_values = [0, limits["HP6633A"]["OVP_lim"]]
+    voltage_values = [0, limits["HP6633A"]["Volt_lim"]]
```

Comment:
In order to have this working, you need to declare the corresponding property with parameter `dynamic=True`.
Same comment for class `HP6634A`

Reply:
Thank you for the reminder, added a `dymaic=True,` to the tree properties

Reply:
I'm very glad that you drove the dynamic property implementation, @msmttchr ! This example shows how this enables really concise model variants without code duplication! :-)